### PR TITLE
Fixing a few rough edges in the GrainTracker

### PR DESCRIPTION
**Pull request by permcody**
refs #11173

### CODE REVIEW

File: modules/phase_field/src/postprocessors/FeatureFloodCount.C [hunk 1]
```diff
@@ -1652,6 +1652,8 @@ FeatureFloodCount::FeatureData::merge(FeatureData && rhs)
   // Logical AND here to combine flags (INACTIVE & INACTIVE == INACTIVE, all other combos are CLEAR)
   _status &= rhs._status;
 
+  _intersects_boundary |= _intersects_boundary;
```
Comment (dschwen) [inline modules/phase_field/src/postprocessors/FeatureFloodCount.C]: Wat?

**Review by dschwen — COMMENTED** [review]
Isn't that a no op?

File: modules/phase_field/src/userobjects/PolycrystalUserObjectBase.C [hunk 2]
```diff
@@ -281,11 +281,15 @@ PolycrystalUserObjectBase::assignOpsToGrains()
...
+
   // Use a simple backtracking coloring algorithm
   if (_coloring_algorithm == "bt")
   {
+    mooseInfo("The backtracking algorithm has exponential complexity. If you are using very few "
+              "order parameters, or you have several hundred grains or more, you should use one of "
+              "the PETSc coloring algorithms such as \"jp\".");
+
     if (!colorGraph(0))
       mooseError("Unable to find a valid grain to op coloring, do you have enough op variables?");
```
[the remark this answers is not in the document]
Comment (dschwen) [reply modules/phase_field/src/userobjects/PolycrystalUserObjectBase.C]: Could we use `paramError` here maybe? And associate it with either the op parameter or the coloring algo parameter?

[the remark this answers is not in the document]
Comment (permcody) [reply modules/phase_field/src/userobjects/PolycrystalUserObjectBase.C]: Good question, I honestly don't know. What I'd need is a paramWarning right? Maybe I don't understand that system yet. I'll look into it.

File: modules/phase_field/src/postprocessors/FeatureFloodCount.C (hunk 1 above)
Comment (permcody) [reply modules/phase_field/src/postprocessors/FeatureFloodCount.C]: This is in the merge method of two partial features. If any part of the feature intersects the boundary, then the merged feature intersects the boundary. I suppose I can add a comment 😄

Comment (dschwen) [reply modules/phase_field/src/postprocessors/FeatureFloodCount.C]: ... are you gas lighting me now? It is the same variable on both sides of the `|=` operator.

```
a = a | a
```
will never change `a`'s value!

Comment (dschwen) [reply modules/phase_field/src/postprocessors/FeatureFloodCount.C]: Did you mean 

```
_intersects_boundary |= rhs._intersects_boundary 
```

maybe?

File: modules/phase_field/src/userobjects/PolycrystalUserObjectBase.C (hunk 2 above)
[the remark this answers is not in the document]
Comment (permcody) [reply modules/phase_field/src/userobjects/PolycrystalUserObjectBase.C]: @dschwen - We don't have a `paramWarning` or `paramInfo` parameter so I can't use that method unless we don't want to allow it at all. (bt is superior at spreading out the grains more evenly and works in a few cases where the PETSc algorithms actually fail (3 ops for a regular hexagonal crystal structure). I have a test for this case checked in! Do you guys want to push for paramInfo and/or paramWarning macros?

File: modules/phase_field/src/postprocessors/FeatureFloodCount.C (hunk 1 above)
Comment (permcody) [reply modules/phase_field/src/postprocessors/FeatureFloodCount.C]: Yes, that's the right one... Oops... I gave this fix to Fande, then mis-typed it here. Perhaps I should work up a test that will fail on 2 procs (current test takes several procs before it fails without this line)

File: modules/phase_field/src/userobjects/PolycrystalUserObjectBase.C (hunk 2 above)
[the remark this answers is not in the document]
Comment (dschwen) [reply modules/phase_field/src/userobjects/PolycrystalUserObjectBase.C]: Oh, I thought we were talking about the _error_ message, but indeed, a `paramInfo` or `paramWarning` would be great for the new statement above it.

[the remark this answers is not in the document]
Comment (amjokisaari) [reply modules/phase_field/src/userobjects/PolycrystalUserObjectBase.C]: I think we have demonstrated one case where we would like a paramWarning/paramInfo type of thing, so I suspect there will be others.  Could be useful in TensorMechanics, for example, or any of these Master Actions that are getting built for modules.

[the remark this answers is not in the document]
Comment (amjokisaari) [reply modules/phase_field/src/userobjects/PolycrystalUserObjectBase.C]: Yes, I was originally talking about the error message "Unable to find..." bla bla.  Thanks for putting us back on track...

[the remark this answers is not in the document]
Comment (permcody) [reply modules/phase_field/src/userobjects/PolycrystalUserObjectBase.C]: @rwcarlsen - Would you add the `paramInfo` and `paramWarning`? I bet there are several existing cases where we can leverage. Thanks!

**Comment by permcody** [comment]
Ping @dschwen - hopefully this is up to your standards.

**Comment by dschwen** [comment]
Do I detect a hint of sarcasm?

File: modules/phase_field/test/tests/grain_tracker_test/tests [hunk 3]
```diff
@@ -16,6 +16,34 @@
...
+
+  [./test_advanced_op_assignment_petsc_error]
+    type = 'RunException'
+    input = 'grain_tracker_advanced_op.i'
+    expect_err = "Unable to find a valid grain to op coloring"
+    recover = false # No solve
+    cli_args = 'UserObjects/voronoi/coloring_algorithm=jp GlobalParams/op_num=7 Outputs/csv=false'
+
+    requirement = "The PolycrystalUserObject base class shall error when a valid coloring cannot be found "
+                  "when using the built-in PETSc based stochastic algorithms."
```
Comment (dschwen) [inline modules/phase_field/test/tests/grain_tracker_test/tests]: _shall_ is great, but can we also use the word _henceforth_?

**Review by dschwen — APPROVED** [review]

File: modules/phase_field/test/tests/grain_tracker_test/tests (hunk 3 above)
Comment (permcody) [reply modules/phase_field/test/tests/grain_tracker_test/tests]: Let's just go Old English and require Iambic pentameter too!

**Pull request merged**